mongodbatlas/private_serverless_endpoints.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+const (
+	serverlessPrivateEndpointsPath = "api/atlas/v1.0/groups/%s/privateEndpoint/serverless/instance/%s/endpoint"
+)
+
+// PrivateEndpointsService is an interface for interfacing with the Private Endpoints
+// of the MongoDB Atlas API.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints/
```

Comment:
```suggestion
// See more: https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Serverless-Private-Endpoints
```

Reply:
@gssbzn Was using docs at the time that stated ... Creates one private endpoint for one Atlas serverless instance. Serverless instances support private endpoints on AWS only. Will have to add support for Azure

Reply:
@martinstibbe Azure support is ok to do as a follow up